plugins/outputs/cloudwatchlogs/cloudwatchlogs_test.go

```diff
@@ -44,3 +49,23 @@ func TestCreateDest(t *testing.T) {
 		t.Errorf("Empty create dest should return dest to default group and stream, %v/%v found", d.pusher.Group, d.pusher.Stream)
 	}
 }
+
+func TestUserAgent(t *testing.T) {
+	c := outputs.Outputs["cloudwatchlogs"]().(*CloudWatchLogs)
+	target := Target{
+		Group:     "/aws/containerinsights/cluster-name/prometheus",
+		Stream:    "Stream",
+		Retention: -1,
+	}
+	d0 := c.getDest(target)
+	logClient := d0.pusher.Service.(*cloudwatchlogs.CloudWatchLogs)
+	req := request.New(
+		aws.Config{}, metadata.ClientInfo{}, logClient.Handlers, nil, &request.Operation{
+			HTTPMethod: "GET",
+			HTTPPath:   "/",
+		}, nil, nil)
+	logClient.Handlers.Build.Run(req)
+	if !strings.Contains(req.HTTPRequest.UserAgent(), "container_insights") {
```

Comment:
Looking at the code change above in the implementation, I don't see it injecting the `container_insights` string into the user agent header. Is this the assertion we want to be making here? Currently the code appends to the `agentinfo.OutputPlugins` array, so I think the most direct assertion we could make is checking that the array contains `container_insights`. Can you explain why we're checking the HTTP request UserAgent value for `container_insights` here? It's not obvious to me how that is being set. 

Further, I think it would be good to add an assertion to an existing test (if we don't want to write another test for this sole purpose) to verify that the `container_insights` value is NOT appended when the group does not match the Regexp that you added.

Reply:
The `container_insights` string is added into the HTTP request header from this line:
```
client.Handlers.Build.PushBackNamed(handlers.NewCustomHeaderHandler("User-Agent", agentinfo.UserAgent()))
```
where `agentinfo.UserAgent()` builds string from `agentinfo.OutputPlugins` and other variable.

But I agree this is not an obvious/direct way for an unit test. @jhnlsn also pointed out we could just checked `agentinfo.UserAgent()`/`agentinfo.OutputPlugins` for the expected values. I've updated the test code and added some more test cases to cover more